File: Chorme_extension/contentScript/All-in-JPEG/content/contentType.js
 const ContentType = {
    Image: 'Image',
    Audio: 'Audio',
    Text: 'Text'
};

 const ContentAttribute = {
    none: { code: 0 },
    basic: { code: 1 },
    burst: { code: 2 },
    object_focus: { code: 3 },
    distance_focus: { code: 4 },
    magic: { code: 5 },
    edited: { code: 6 },

    fromCode: function(code) {
        for (const attr in this) {
            if (this[attr].code === code) {
                return attr;
            }
        }
        return this.none;
    }
};
export default ContentAttribute;

File: OnePICDiary/app/src/main/java/com/goldenratio/onepicdiary/PictureModule/Contents/ContentType.kt
package com.goldenratio.onepic.AllinJPEGModule.Contents

enum class ContentType {
    Image,
    Audio,
    Text
}

enum class ContentAttribute(val code: Int ) {
    none(0),
    Basic(1),
    Burst(2),
    object_focus(3),
    distance_focus(4),
    magic(5),
    edited(6);

    companion object {
        fun fromCode(code: Int) = values().firstOrNull { it.code == code } ?: none
    }
}
